File: examples/mploed-applicant-scoring/src/test/java/io/example/asyncapi/provider/applicantscoring/config/RepositoriesInMemoryConfig.java
package io.example.asyncapi.provider.applicantscoring.config;

import io.zenwave360.example.applicantscoring.repository.jpa.*;
import io.zenwave360.example.applicantscoring.repository.jpa.inmemory.*;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;

// @Configuration
public class RepositoriesInMemoryConfig {

    protected final ApplicantScoringRepository applicantScoringRepository = new ApplicantScoringRepositoryInMemory();

    @Bean
    @Primary
    public <T extends ApplicantScoringRepository> T applicantScoringRepository() {
        return (T) applicantScoringRepository;
    }
}

File: examples/order-fulfillment-kotlin/src/test/kotlin/io/zenwave360/example/orderfulfillment/config/RepositoriesInMemoryConfig.kt
package io.zenwave360.example.orderfulfillment.config

import io.zenwave360.example.orderfulfillment.*
import io.zenwave360.example.orderfulfillment.inmemory.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary

// @Configuration
open class RepositoriesInMemoryConfig {

    protected val orderRepository = OrderRepositoryInMemory()

    @Bean
    @Primary
    fun orderRepository(): OrderRepositoryInMemory {
        return orderRepository
    }
}
